File: template-for-repurposing/culture-monitor-app/tmp/cleanup_typography.py
import os
import re

def clean_classes(content):
    # Regex for font-black, uppercase, and tracking-* (NOT tight/tighter)
    # The [^\w-] ensures we don't match sub-words, or it's the end of the string
    
    # 1. Remove font-black
    content = re.sub(r'(?<=\s)font-black(?=\s|$)|(?<=")font-black(?=\s|")|(?<=\s)font-black(?=")', '', content)
    
    # 2. Remove uppercase
    content = re.sub(r'(?<=\s)uppercase(?=\s|$)|(?<=")uppercase(?=\s|")|(?<=\s)uppercase(?=")', '', content)
    
    # 3. Remove tracking-* except tracking-tight and tracking-tighter
    # We use a negative lookahead to exclude tight and tighter
    content = re.sub(r'\btracking-(?!(tight|tighter)\b)[a-z\[\]0-9.-]+\b', '', content)
    
    # 4. Cleanup double spaces and spaces near quotes in className attributes
    # Find all className="..." or className={`...`} etc.
    def cleanup_spacing(match):
        attr = match.group(0)
        # Remove extra spaces
        cleaned = re.sub(r'\s{2,}', ' ', attr)
        # Remove space after " or {`
        cleaned = re.sub(r'="\s+', '="', cleaned)
        cleaned = re.sub(r'={`\s+', '={`', cleaned)
        # Remove space before " or `}
        cleaned = re.sub(r'\s+"', '"', cleaned)
        cleaned = re.sub(r'\s+`}', '`}', cleaned)
        return cleaned

    content = re.sub(r'className=(("[^"]*")|( {`[^`]*`})|( {[^}]*}))', cleanup_spacing, content)
    
    return content

def process_directory(directory):
    for root, dirs, files in os.walk(directory):
        for file in files:
            if file.endswith(('.tsx', '.ts')):
                path = os.path.join(root, file)
                try:
                    with open(path, 'r', encoding='utf-8') as f:
                        content = f.read()
                    
                    new_content = clean_classes(content)
                    
                    if new_content != content:
                        with open(path, 'w', encoding='utf-8') as f:
                            f.write(new_content)
                        print(f"Cleaned: {path}")
                except Exception as e:
                    print(f"Error processing {path}: {e}")

if __name__ == "__main__":
    process_directory("c:\\wamp64\\www\\culture-monitor-app\\frontend\\src")

File: template-for-repurposing/culture-monitor-app/tmp/cleanup_typography.js
const fs = require('fs');
const path = require('path');

function cleanClasses(content) {
    // 1. Remove font-black
    content = content.replace(/\bfont-black\b/g, '');
    
    // 2. Remove uppercase
    content = content.replace(/\buppercase\b/g, '');
    
    // 3. Remove tracking-* except tight and tighter
    content = content.replace(/\btracking-(?!(tight|tighter)\b)[a-z\[\]0-9.-]+\b/g, '');
    
    // 4. Specifically target CardHeader components and remove bg-muted/30, bg-muted/50, and border-b
    // We look for CardHeader followed by a className attribute
    content = content.replace(/(<CardHeader[^>]*className=(?:"([^"]*)"|{`([^`]*)`})[^>]*>)/g, (match, p1, p2, p3) => {
        let classes = p2 || p3 || "";
        // Remove the specific header classes
        let cleaned = classes.replace(/\b(bg-muted\/30|bg-muted\/50|border-b)\b/g, '').replace(/\s{2,}/g, ' ').trim();
        if (p2 !== undefined) return p1.replace(`className="${p2}"`, `className="${cleaned}"`);
        if (p3 !== undefined) return p1.replace(`className={\`${p3}\`}`, `className={\`${cleaned}\`}`);
        return match;
    });

    // 5. General cleanup of double spaces and spaces near quotes/brackets in className attributes
    content = content.replace(/className=(?:"([^"]*)"|{`([^`]*)`})/g, (match, p1, p2) => {
        let classes = p1 || p2 || "";
        let cleaned = classes.replace(/\s{2,}/g, ' ').trim();
        if (p1 !== undefined) return `className="${cleaned}"`;
        if (p2 !== undefined) return `className={\`${cleaned}\`}`;
        return match;
    });

    return content;
}

function processDirectory(dir) {
    const files = fs.readdirSync(dir);
    for (const file of files) {
        const filePath = path.join(dir, file);
        const stat = fs.statSync(filePath);
        if (stat.isDirectory()) {
            processDirectory(filePath);
        } else if (filePath.endsWith('.tsx') || filePath.endsWith('.ts')) {
            try {
                const content = fs.readFileSync(filePath, 'utf-8');
                const newContent = cleanClasses(content);
                if (newContent !== content) {
                    fs.writeFileSync(filePath, newContent, 'utf-8');
                    console.log(`Cleaned: ${filePath}`);
                }
            } catch (err) {
                console.error(`Error processing ${filePath}: ${err.message}`);
            }
        }
    }
}

processDirectory('c:\\wamp64\\www\\culture-monitor-app\\frontend\\src');
